app/guestbook/page.tsx

```diff
@@ -1,30 +1,65 @@
-import { auth } from 'app/auth';
+import { getSession } from 'app/auth';
 import { getGuestbookEntries } from 'app/db/queries';
-import { SignIn, SignOut } from './buttons';
-import { Suspense } from 'react';
+import { logout } from 'app/auth';
+import { redirect } from 'next/navigation';
 import Form from './form';
 
 export const metadata = {
   title: 'Guestbook',
   description: 'Sign my guestbook and leave your mark.',
 };
 
+function SignOut() {
+  return (
+    <form
+      action={async () => {
+        'use server';
+        await logout();
+        redirect('/guestbook');
+      }}
+    >
+      <button
+        className="text-xs text-neutral-700 dark:text-neutral-300 mt-2 mb-6"
+        type="submit"
+      >
+        Log out
+      </button>
+    </form>
+  );
+}
+
+function SignIn() {
+  return (
+    <a
+      className="px-3 py-2 border border-neutral-200 dark:border-neutral-700 bg-neutral-50 dark:bg-neutral-800 rounded p-1 text-sm inline-flex items-center leading-4 text-neutral-900 dark:text-neutral-100 mb-8"
+      href={`https://vercel.com/api/login/oauth/authorize?client_id=${
+        process.env.OAUTH_CLIENT_ID
+      }&redirect_uri=${encodeURIComponent(
+        process.env.OAUTH_REDIRECT_URI!
+      )}&response_type=code&scope=openid+email+profile&state=${Math.random()
+        .toString(36)
+        .substring(7)}`}
```

Comment:
`state` should be generated using a cryptographically random value, eg:

```ts
function getRandomState() {
  const CHUNK_SIZE = 0x8000
  const input = crypto.getRandomValues(new Uint8Array(32))
  const arr = []
  for (let i = 0; i < input.byteLength; i += CHUNK_SIZE) {
    arr.push(String.fromCharCode.apply(null, input.subarray(i, i + CHUNK_SIZE)))
  }
  return btoa(arr.join('')).replace(/=/g, '').replace(/\+/g, '-').replace(/\//g, '_')
}
```

It should also be saved in a cookie for verification later on at https://github.com/leerob/leerob.io/pull/719/files#r1633836953